routes/actions.js

```diff
@@ -343,25 +344,52 @@ router.post('/register/:code', function(req, res, next) {
 
 });
 
-function sendRegistrationForm(req, res, formInfo) {
+async function sendRegistrationForm(req, res, formInfo) {
   let pageErrors = req.flash('pageErrors');
 
   const { code } = req.params;
-
-  render.template(req, res, 'register', {
+  const signupTeams = req.query.signupTeams;
+  // checks if signupTeams exist, valid teams are passed to render
+  if (signupTeams) {
+    let signupTeamsArray = signupTeams.split(' ');
+    let validSignupTeams = [];
+    let promise_array = [];
+    for (let i = 0; i < signupTeamsArray.length; i++) {
+      let team = slugs.resolveAndLoadTeam(req, res, signupTeamsArray[i])
```

Comment:
It may be more appropriate to use a method from team-provider.js as you originally suggested.  I used slugs instead because team-provider methods were returning errors about missing arguments when I tried to pass the slug alone.